modules/runner/src/main/java/org/apache/ignite/internal/app/IgniteImpl.java

```diff
@@ -464,6 +463,13 @@ public class IgniteImpl implements Ignite {
                 localConfigurationValidator
         );
 
+        // Start local configuration to be able to read all local properties.
+        try {
+            lifecycleManager.startComponentsAsync(new ComponentContext(), nodeCfgMgr);
```

Comment:
Wow, I'm amazed that local configuration manager has absolutely no dependencies, that's wild!

Could there be any issues with passing a different context instance? Maybe it would affect stopping somehow? That's just a question, I'm not aware of what this context is doing